JASP-Engine/JASP/R/commonMachineLearningClustering.R

```diff
@@ -115,7 +120,16 @@
 
   clusteringTable$addCitation("Hartigan, J. A., & Wong, M. A. (1979). Algorithm AS 136: A k-means clustering algorithm. Journal of the Royal Statistical Society. Series C (Applied Statistics), 28(1), 100-108.")
 
-  if(!ready) clusteringTable$addFootnote(message="Please provide at least 2 variables.", symbol="<i>Note.</i>")
+  if(type != "densitybased"){
+    clusters  <- base::switch(options[["modelOpt"]], "validationManual" = options[["noOfClusters"]], "validationOptimized" = options[["maxClusters"]])
+    tooMuch   <- clusters > (nrow(dataset) - 1)
+
+    if(!ready && tooMuch)
+      clusteringTable$addFootnote(message = "You have specified more clusters than distinct data points", symbol="<b>Warning.</b>")
```

Comment:
This is a user input error -- like when entering infinity or zero variance, etc. Shouldn't this be handled by `.hasErrors()` as is the custom?

Reply:
So something like
```
.errorHandlingClusteringAnalyses <- function(dataset, options, type) {
  predictors <- unlist(options$predictors)
  if (length(predictors[predictors != ""]) == 0L)
    return()

  checkClusters <- function() {
    if (type != "densitybased") {
      clusters  <- base::switch(options[["modelOpt"]], "validationManual" = options[["noOfClusters"]], "validationOptimized" = options[["maxClusters"]])
      if (clusters > (nrow(dataset) - 1))
        return("You have specified more clusters than distinct data points")
    }
  }
  
  .hasErrors(dataset, type = c('infinity', 'observations'), custom=checkClusters, all.target = predictors,
            observations.amount = "< 2", exitAnalysisIfErrors = TRUE)

  return()
}
```

where the function now requires `type`

Reply:
And if you set `ready` before you called `.errorHandlingClusteringAnalyses` you would not need to check the number of predictors

Reply:
Hmm, point is that JASP crashes quite hard when you do this right? Isn't it better that we just show empty tables?